src/main/java/org/elasticsearch/app/api/server/services/ConfigManager.java

```diff
@@ -122,6 +122,31 @@ public void addUpdateRecordToIndex(String indexName, UpdateRecord updateRecord)
         riverDAO.save(river);
     }
 
+    @Transactional(readOnly = true)
+    public List<Map<String, Object>> getAllConfigs() {
+        return riverDAO.findAll().stream().map(River::toMap).collect(Collectors.toList());
+    }
+
+    @Transactional()
+    public List<Integer> setAllConfigs(List<String> configs) {
+        int newConfigs = 0;
+        int updatedConfigs = 0;
+        for (String config : configs) {
+            River newRiver = createRiver(config);
```

Comment:
Isn't this in fact em.merge?

Reply:
It is. Last line in `for` does it `riverDAO.save(foundRiver);` But to retrieve how many was added as new I need to know if it exists => same problem as the next comment.

Reply:
Changed to a reference. Merged by save().